### PR TITLE
[DO NOT MERGE] Refactor content-host katello-agent tests and add more coverage

### LINKED ISSUE #1709: Add CLI test for content-host package and sub commands
```
hammer content-host package --help

Usage:
    hammer content-host package [OPTIONS] SUBCOMMAND [ARG] ...

Parameters:
 SUBCOMMAND                    subcommand
 [ARG] ...                     subcommand arguments

Subcommands:
 install                       Install packages remotely
 remove                        Uninstall packages remotely
 upgrade                       Update packages remotely
 upgrade-all                   Update packages remotely

Options:
 -h, --help                    print help
```

### PR DESCRIPTION
**Pull request by abalakh**
1. Replaced `options.get('foo', make_foo())` with `if ... else` statements. Short explanation:
    ```
>>> def foo():
...     print 'foo'
...     return 'foo [2]'
...
>>> bar = {'test': 'bar'}
>>> bar.get('test', foo())
foo
'bar'
```
    Function, passed as `default=` value, was always executing, so each test was creating a lot of redundant entities.

2. Updated tests logic. Before we had something like this
    ```
def setUp():
# Enable RHEL repo
# Create activation key with subscription to repo with katello-agent
# Register content-host

def test_...():
# Create custom repo
# Update activation key with subscription to repo
# Test things
```
    As per one email discussion, updating activation key with new subscription doesn't update subscriptions on existing content host, updated activation key will only affect newly registered content hosts. So creation of custom repo and updating of activation key was moved _before_ content-host registration.
    In addition, as all the tests were creating custom repo for testing purposes, so i moved custom repo, upd of AK & registering of content-host to setUp section, which should add readability.

3. Added tests for remote package install/remove/upgrade.
    Closes #1709

    ```
nosetests tests/foreman/cli/test_contenthost.py -m test_contenthost_
......
----------------------------------------------------------------------
Ran 6 tests in 2188.351s

OK
```

### CODE REVIEW

**Comment by abalakh** [comment]
`[DO NOT MERGE]` as
1. It depends on SatelliteQE/automation-tools#211
2. It's not enough to simply merge SatelliteQE/automation-tools#211 , our jenkins machines should install Sat6 with that fix, and AFAIK machines are reinstalled once a week. Without that fix newly added tests will fail.
3. (Not sure) PR may conflict with #2633 , as it applies part of #2633 fix (without that, tests wouldn't work so i couldn't verify the tests). Maybe one of PRs should be rebased after another one gets merged.

**Comment by Ichimonji10** [comment]
See SatelliteQE/nailgun#171.

File: robottelo/cli/factory.py [hunk 1]
```diff
@@ -1711,10 +1712,10 @@ def setup_org_for_a_custom_repo(options=None):
     if result.return_code != 0:
         raise CLIFactoryError('Failed to synchronize repository')
     # Create CV if needed and associate repo with it
-    cv_id = options.get(
-        'content-view-id',
-        make_content_view({u'organization-id': org_id})['id']
-    )
+    if options.get('content-view-id') is None:
```
Comment (elyezer) [inline robottelo/cli/factory.py]: cv_id = options.get(
    'content-view-id',
    make_content_view({u'organization-id': org_id})['id']
)

Comment (abalakh) [inline robottelo/cli/factory.py]: It doesn't work the way you expect it to work. `make_content_view({u'organization-id': org_id})['id']` will be ran, received `id` will be set to `default=` argument and then `options.get('content-view-id', default=*id*)` will be executed. As a result content view will always be created, even if `'content-view-id'` is not None. So it's not an option to use `get()` in such cases.

File: robottelo/cli/factory.py [hunk 2]
```diff
@@ -1694,11 +1692,14 @@ def setup_org_for_a_custom_repo(options=None):
             not options.get('url', None)):
         raise CLIFactoryError('Please provide valid custom repo URL.')
     # Create new organization and lifecycle environment if needed
-    org_id = options.get('organization-id', make_org()['id'])
-    env_id = options.get(
-        'lifecycle-environment-id',
-        make_lifecycle_environment({u'organization-id': org_id})['id']
-    )
+    if options.get('organization-id') is None:
```
Comment (elyezer) [inline robottelo/cli/factory.py]:
```python
org_id = options.get(
    'organization-id',
    make_org()['id']
)
```

File: robottelo/cli/factory.py [hunk 3]
```diff
@@ -1694,11 +1692,14 @@ def setup_org_for_a_custom_repo(options=None):
...
+        org_id = make_org()['id']
+    else:
+        org_id = options['organization-id']
+    if options.get('lifecycle-environment-id') is None:
```
Comment (elyezer) [inline robottelo/cli/factory.py]:
```python
env_id = options.get(
    'lifecycle-environment-id',
    make_lifecycle_environment({u'organization-id': org_id})['id']
)
```

File: robottelo/cli/factory.py [hunk 4]
```diff
@@ -1738,14 +1739,14 @@ def setup_org_for_a_custom_repo(options=None):
     if result.return_code != 0:
         raise CLIFactoryError('Failed to promote version to next environment')
     # Create activation key if needed and associate content view with it
-    if 'activationkey-id' not in options:
+    if options.get('activationkey-id') is None:
```
Comment (elyezer) [inline robottelo/cli/factory.py]: activationkey_id = options.get(
    'activationkey-id',
    make_activation_key({
        u'content-view-id': cv_id,
        u'lifecycle-environment-id': env_id,
        u'organization-id': org_id,
    })['id']
)

File: robottelo/cli/factory.py [hunk 5]
```diff
@@ -1798,17 +1798,20 @@ def setup_org_for_a_rh_repo(options=None):
...
+            not options.get('repository')):
         raise CLIFactoryError(
             'Please provide valid product, repository-set and repo.')
     # Create new organization and lifecycle environment if needed
-    org_id = options.get('organization-id', make_org()['id'])
-    env_id = options.get(
-        'lifecycle-environment-id',
-        make_lifecycle_environment({u'organization-id': org_id})['id']
-    )
+    if options.get('organization-id') is None:
```
Comment (elyezer) [inline robottelo/cli/factory.py]: Take advantage of the get second parameter here, like previous comments.

File: robottelo/cli/factory.py [hunk 6]
```diff
@@ -1798,17 +1798,20 @@ def setup_org_for_a_rh_repo(options=None):
...
+        org_id = make_org()['id']
+    else:
+        org_id = options['organization-id']
+    if options.get('lifecycle-environment-id') is None:
```
Comment (elyezer) [inline robottelo/cli/factory.py]: Take advantage of the get second parameter here, like previous comments.

File: robottelo/cli/factory.py [hunk 7]
```diff
@@ -1820,34 +1823,34 @@ def setup_org_for_a_rh_repo(options=None):
...
+        u'product': options['product'],
     })
     if result.return_code != 0:
         raise CLIFactoryError('Failed to synchronize repository')
     # Create CV if needed and associate repo with it
-    cv_id = options.get(
-        'content-view-id',
-        make_content_view({u'organization-id': org_id})['id']
-    )
+    if options.get('content-view-id') is None:
```
Comment (elyezer) [inline robottelo/cli/factory.py]: Take advantage of the get second parameter here, like previous comments.

File: robottelo/cli/factory.py [hunk 8]
```diff
@@ -1871,14 +1874,14 @@ def setup_org_for_a_rh_repo(options=None):
     if result.return_code != 0:
         raise CLIFactoryError('Failed to promote version to next environment')
     # Create activation key if needed and associate content view with it
-    if 'activationkey-id' not in options:
+    if options.get('activationkey-id') is None:
```
Comment (elyezer) [inline robottelo/cli/factory.py]: Take advantage of the get second parameter here, like previous comments.

File: robottelo/cli/factory.py (hunk 1 above)
Comment (elyezer) [inline robottelo/cli/factory.py]: You are completely right @abalakh :+1:

**Comment by elyezer** [comment]
Closing this in favor of #2659

**Pull request closed without merging**